### DISCUSSION TITLE
In what circumstances would an InlineFragment not have a TypeCondition

### DISCUSSION THREAD

**Discussion by JesalR** [graphql/graphql-js#3975] [Q&A]
Hi,

I'm writing a transformer for incoming user queries to remove certain nodes if they do not have permission to access them. 

I'm having to figure that path of each accessed variable to map back to the schema to check if it has a specific directive there.

If one of the selection nodes is an inline fragment, I'm seeing that the `typeCondition` member is optional, but in all descriptions of the language I've seen online, I cannot find a case where this might happen. Does anyone happen to know the cases where it could be undefined?

**Comment by official-burak** [comment]
When `on Type` is omitted. `... { field }` and `... @include(if: $x) { field }` apply to the parent type, so `typeCondition` is undefined.

The parser only sets it after the `on` keyword:

```ts
typeCondition: hasTypeCondition ? this.parseNamedType() : undefined
```

`...FragName` is a `FRAGMENT_SPREAD`, not an inline fragment with a missing type.
